pkg/api/validation/validation.go

```diff
@@ -2272,6 +2276,12 @@ func ValidatePodSpec(spec *api.PodSpec, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, ValidateHostAliases(spec.HostAliases, fldPath.Child("hostAliases"))...)
 	}
 
+	if len(spec.PriorityClassName) > 0 {
```

Comment:
Added the feature gate for PriorityClassName. Admission controller (a subsequent PR) will prevent users from setting `Priority` regardless. `Priority` is set only be the admission controller.